main.py

```diff
@@ -53,7 +53,131 @@ def handle_arguments(cl_arguments):
     return parser.parse_args(cl_arguments)
 
 
-def _try_logging_git_info():
+def setup_target_task_training(args, target_tasks, model):
+    """
+    Save model states from pretrianing if applicable, and 
+    loads the correct model state for the target task training 
+    stage.
+    
+    Parameters
+    ----------------
+    args: Params object
+    target_tasks: list of target Task objects
+    mdoel: a MultTaskModel object
+
+    Returns
+    ----------------
+    task_names_to_avoid_loading: list of strings, if we don't allow for 
+    use of pretrained target specific module parameters, then this list will 
+    cosnist of all the task names so that we avoid loading the 
+    pretrained parameters. Else, it will be an empty list.
+    strict: bool, indicates whether to exit if the loaded model has some expected
```

Comment:
This name is a bit weird - it seems like something you should be passing _to_ this function, not returning.

Looking at the code below, the control flow is also a bit odd - since this value can be computed from `args` alone, would it be cleaner to just compute outside in the calling code and pass as an argument here?

Reply:
I agree. Not strongly opposed to merging this way, but it's definitely not making the code more readable/editable.